### PR TITLE
tough: blanket impl sign for references

### PR DESCRIPTION
**Pull request by webern**
*Issue #, if available:*

#446

*Description of changes:*

Implement `Sign` for references of types that implement `Sign`.
@iliana is this what's needed?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tough/src/sign.rs [hunk 1]
```diff
@@ -27,6 +28,21 @@ pub trait Sign: Sync + Send {
...
+impl<'a, T: Sign> Sign for &'a T {
+    fn tuf_key(&self) -> Key {
+        (*self).tuf_key()
+    }
+
+    fn sign(
+        &self,
+        msg: &[u8],
+        rng: &dyn SecureRandom,
+    ) -> std::prelude::rust_2015::Result<Vec<u8>, Box<dyn Error + Send + Sync + 'static>> {
```
Comment (iliana) [inline tough/src/sign.rs:41]:
```suggestion
    ) -> std::result::Result<Vec<u8>, Box<dyn Error + Send + Sync + 'static>> {
```

what happened here 😅

**Review by iliana — APPROVED** [review]

File: tough/src/sign.rs (hunk 1 above)
Comment (webern) [reply tough/src/sign.rs:41]: Weird. Maybe fmt did that

Comment (zmrow) [reply tough/src/sign.rs:41]: :smile:

**Review by zmrow — APPROVED** [review]
:rabbit:

**Pull request merged**